meddocan/language/method_extensions.py

```diff
@@ -99,14 +100,17 @@ def get_line(token: Token) -> str:
                 if write_sentences:
                     line = "\n"
                 else:
-                    line = "\\n O"
+                    line = "\\n O\n"
                 lines.append(line)
 
             if write_sentences:
                 f.writelines(lines)
             else:
                 joined_lines = "".join((*lines, "\n"))
                 f.write(joined_lines)
+            if document_separator_token is not None:
+                document_separator_line = f"{document_separator_token} O\n\n"
+                f.write(document_separator_line)
```

Comment:
In ``flair/datasets/sequence_labeling.ColumnDataSet._convert_lines_to_sentence`` line 668

```python
        # check if this sentence is a document boundary
        if sentence.to_original_text() == self.document_separator_token:
            sentence.is_document_boundary = True
```

This is then use to determine if a ``Flair.data.Sentence`` as a left or right context. Cf ``Flair.data.Sentence.left_context`` and ``lFlair.data.Sentence.right_context

The ``flair.data.Sentence.to_original_text`` just return the orignal raw text. TODO verify this in a test!